Scenarios/BlueGreen-Deployment-for-AKS/GAP/bluegreen-aks-solution-content.md

```diff
@@ -0,0 +1,222 @@
+
+There are two Azure Servies listed in the [Componets Section](#components), for which is possible to use also alternative products and/or OSS solutions. The two services are:
+- [Application Gateway](https://azure.microsoft.com/services/application-gateway/)
+- [Container Registry](https://azure.microsoft.com/services/container-registry/)
+
+The intent of the article is not to provide a curated list of alternatives, but is to emphasize that is possible to adopt different components to achieve the same implementation of the patern.
+Just to give some example possible alternatives are:
+- NGINX, HAProxy, etc.. instead of Application Gateway
+- Harbor, etc.. instead of Container Registry
+
+The following alternative solutions provide scenario-focused lenses to build off of this core architecture: 
+
+- [Link to first solution idea or other architecture that builds off this solution](filepath.yml)
+- [Second solution idea that builds off this solution](filepath.yml)
+
+
+## Considerations
+The following considerations have their basis on the [Microsoft Azure Well-Architected Framework](/azure/architecture/framework) and [Cloud Adoption Framework](addlink).
+One main consideration is that this pattern can be adopted in a full automated scenario, like a zero touch deployment. Usually the basic implementation has a manual trigger to activate the different steps described. Along the way and with the proper maturity and monitoring features is possible to automate also the triggers, that means that there are automated testing and specific metrics, SLA and SLO to automatize the triggers.
+
+One important lesson learned is to have dedicated hostnames for the blue green clusters and also have a dedicated endpoints configuration on the Gateway/Load Balancer in front of the clusters. This is critical to improve the reliability and validity of the deployment of the new cluster. In this way the validation of the deployment happen with the same architecture and configurations of a standard production cluster.
+
+The blue green pattern is also an enabler to implement and test BC/DR solutions for AKS and related workloads.
+
+A key aspect is related to the **Cost Optimization**, in detail the blue green cluster pattern is widely adopted in the cloud for the native elasticiy provided by cloud providers, in particular Azure, this include the cost saving and control to manage and mantain the blue green cluster deploymentm; in paritcular as described the main action is to destroy the cluster that is no longer needed.
+
+It is important to highlight that a succesfull implementation of the pattern is related to the fact that the all the aspects like automation, monitoring and validation need to be applied at AKS Platform levele but also the workloads/apps deployed on top of it; only with the E2E coverge is possible to really benefit of the value of the blue green pattern.
```

Comment:
Revise this sentence as it is not easy to understand

Reply:
Also the statement at line @160 is not clear?
@bahramr